src/library-authoring/create-collection/CreateCollectionModal.tsx

```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+import {
+  ActionRow,
+  Button,
+  Form,
+  ModalDialog,
+} from '@openedx/paragon';
+import { useParams } from 'react-router-dom';
+import { useIntl } from '@edx/frontend-platform/i18n';
+import { Formik } from 'formik';
+import * as Yup from 'yup';
+import FormikControl from '../../generic/FormikControl';
+import { LibraryContext } from '../common/context';
+import messages from './messages';
+import { useCreateLibraryCollection } from '../data/apiHooks';
+import { ToastContext } from '../../generic/toast-context';
+
+const CreateCollectionModal = () => {
+  const intl = useIntl();
+  const { libraryId } = useParams();
+  const create = useCreateLibraryCollection(libraryId);
+  const {
+    isCreateCollectionModalOpen,
+    closeCreateCollectionModal,
+  } = React.useContext(LibraryContext);
+  const { showToast } = React.useContext(ToastContext);
+
+  const [isCreatingCollection, setIsCreatingCollection] = React.useState<boolean>(false);
+
+  const handleOnClose = React.useCallback(() => {
+    closeCreateCollectionModal();
+    setIsCreatingCollection(false);
+  }, []);
+
+  const handleCreate = React.useCallback((values) => {
+    setIsCreatingCollection(true);
+
+    create.mutateAsync(values).then(() => {
+      handleOnClose();
+      showToast(intl.formatMessage(messages.createCollectionSuccess));
+    }).catch(() => {
+      setIsCreatingCollection(false);
+      showToast(intl.formatMessage(messages.createCollectionError));
+    });
+  }, []);
+
+  return (
+    <ModalDialog
+      title={intl.formatMessage(messages.createCollectionModalTitle)}
+      isOpen={isCreateCollectionModalOpen}
+      onClose={handleOnClose}
+      size="xl"
+      hasCloseButton
+      isFullscreenOnMobile
+    >
+      <ModalDialog.Header>
+        <ModalDialog.Title>
+          {intl.formatMessage(messages.createCollectionModalTitle)}
+        </ModalDialog.Title>
+      </ModalDialog.Header>
+
+      <Formik
+        initialValues={{
+          title: '',
+          description: '',
+        }}
+        validationSchema={
+          Yup.object().shape({
+            title: Yup.string()
+              .required(intl.formatMessage(messages.createCollectionModalNameInvalid)),
+            description: Yup.string(),
+          })
+        }
+        onSubmit={(values) => handleCreate(values)}
+      >
+        {(formikProps) => (
+          <>
+            <ModalDialog.Body className="mw-sm">
+              <Form onSubmit={formikProps.handleSubmit}>
+                <FormikControl
+                  name="title"
+                  label={(
+                    <Form.Label className="font-weight-bold h3">
+                      {intl.formatMessage(messages.createCollectionModalNameLabel)}
+                    </Form.Label>
+                  )}
+                  value={formikProps.values.title}
+                  placeholder={intl.formatMessage(messages.createCollectionModalNamePlaceholder)}
+                  help=""
```

Comment:
Nit
```suggestion
```

Reply:
![image](https://github.com/user-attachments/assets/d3b29f84-cb99-477a-b230-ddb104f6153c)

I get this error

Reply:
I don't understand.. [`help` has a defaultProp](https://github.com/openedx/frontend-app-course-authoring/pull/1259/files#diff-ee226d1c0d8fcb7863a32835bdf13eb556d3299b6cb6e71da4c4806a1fd9ed9cR55) -- why isn't it ok to remove it?

But if it must be included, then just send an empty element so we don't have to change the expected types:
```suggestion
                  help=<></>
```